service/controller/app/v1/resource/chart/desired_test.go

```diff
@@ -22,7 +21,7 @@ func Test_Resource_GetDesiredState(t *testing.T) {
 		obj           *v1alpha1.App
 		appCatalog    v1alpha1.AppCatalog
 		expectedChart *v1alpha1.Chart
-		errorMatcher  func(error) bool
```

Comment:
Change it to simple bool type so we don't have to match against `executionFailedError`